### ISSUE TITLE
Cedrus X Cedar

### ISSUE THREAD

**Issue by wolfallein** [aodzip/cedar#3]
Have you compared both video encoder/decoder on v3s?

I'm trying to get cedrus to work on my v3s board. I tried to use your dts (adapted to Cedrus), but don't seem to work. Reading the v3s datasheet it says that the sram_c address is not the same address as the H3, or A10 boards... So maybe the ve_sram address is not right., but if you managed to get Cedar working with those addresses it should work for Cedrus as well, right?

Another question is what you have in your system? Could you send your defconfig? What is the OS you using? What software are you using to test? ffmpeg?

Since v3s has very limited ram, what kind of os video are you managing to have working?

**Comment by aodzip** [comment]
I start this project because Cedrus does not run well on S3/V3s.
I used to run Cedrus successfully, but it outputs a lot of mosaics on video.

SRAM_C address is not important. LOL.
Cedrus, Cedar, even VE itself, won't care about the address in dts.
Allocate the SRAM_C to VE is just write 0x7fffffff in SRAM Controller register.

**Comment by wolfallein** [comment]
Thanks for replying 😄 

Maybe there is something wrong with my system because I get a kernel crash when I try to load the `sunxi_cedrus` module. I don't even get  the `/dev/video` / `dev/media` populated.

I tried Cedrus with another board (Allwinner R16) and works nicely up to 720p, but 1080p I get the mosaic thing as well. To see the video I use VLC, and X11.

For the v3s I'm using kernel 5.8 with a custom Debian installation.  What are you using to play the video? The v3s has a very limited memory that x11 and vlc seem very heavy for the soc, but Cedrus only supports vlc for playback.

Would your version support direct rendering to the screen without x11?

I'll try a bit more on the weekend, I'll try to use your driver as well. Seems I need some binaries from Allwinner to get your driver working, where can I get that? Thanks for your work!

BR

**Comment by aodzip** [comment]
A ffmpeg enabled vaapi can test Cedrus.
If you need help with Cedrus, I suggest you go to Cedrus repo for help.

Binary libs here: https://github.com/aodzip/libcedarc

**Comment by wolfallein** [comment]
Ok,

I compiled the kernel with your CedarX driver, but ion driver doesn't compile because I'm trying on kernel 5.8.

I also managed to compile your library using buildroot, but I don't even know how to use it with ffmpeg to try out.

I suspect something like this won't work:

````
ffmpeg -hwaccel vaapi -hwaccel_device /dev/video0 -hwaccel_output_format vaapi -i 
big_buck_bunny_720_stereo.mp4 -f null /dev/null
````
What should I change?

Would be possible to have your kernel compiled (zImage + dtb)? I believe you are using licheepi v3s, so it should be ok. The only extra thing I have is that I'm using the dock and my lcd is the 480x272.

I'm sorry to ask for it. I already spent a lot of time trying to get it to work + a lot of time on cedrus.
